test/numa/run.sh

```diff
@@ -419,6 +472,17 @@ create() { # script API
     done
 }
 
+reset() { # script API
+    # Usage: reset counters
+    #
+    # Resets counters
+    if [ "$1" == "counters" ]; then
+        kind_count[pod]=0
+    else
+        error "invalid reset \"$1\""
+    fi
+}
+
```

Comment:
Can we have something like this, to automatically drop into interactive() at the end of each verify if/as the user asks for it ?
```
diff --git a/test/numa/run.sh b/test/numa/run.sh
index 966990ee..d7c0f013 100755
--- a/test/numa/run.sh
+++ b/test/numa/run.sh
@@ -54,6 +54,11 @@ usage() {
     echo "             0: leave VM running (the default)"
     echo "             1: delete VM"
     echo "             2: stop VM, but do not delete it."
+    echo "    interactive_verify: if/when to go automatically interactive in verify():"
+    echo "             always: go interactive regardless the result of verification"
+    echo "             fail: go interactive if verification fails"
+    echo "             never: never go interactive"
+    echo "             The default is never."
     echo ""
     echo "  Test input VARs:"
     echo "    topology: JSON to override NUMA node list used in tests."
@@ -361,12 +366,19 @@ verify() { # script API
     get-py-cache
     for py_assertion in "$@"; do
         speed=1000 out "### Verifying assertion '$py_assertion'"
-        ( speed=1000 pyexec "assert(${py_assertion})" ) || {
+        if ! ( speed=1000 pyexec "assert(${py_assertion})" ); then
                 out "### The assertion FAILED"
                 echo "verify: assertion '$py_assertion' failed." >> "$SUMMARY_FILE"
-                command-exit-if-not-interactive
-        }
-        speed=1000 out "### The assertion holds."
+                case "$interactive_verify" in
+                    "always"|"1"|"true"|"fail"*) interactive;;
+                    *) command-exit-if-not-interactive;;
+                esac
+        else
+            speed=1000 out "### The assertion holds."
+                case "$interactive_verify" in
+                    "always"|"1"|"true") interactive;;
+                esac
+        fi
     done
 }
 
@@ -515,6 +527,7 @@ vm=${vm-"crirm-test-numa"}
 cri_resmgr_cfg=${cri_resmgr_cfg-"${SCRIPT_DIR}/cri-resmgr-memtier.cfg"}
 cleanup=${cleanup-0}
 reinstall_cri_resmgr=${reinstall_cri_resmgr-0}
+interactive_verify=${interactive_verify-never}
 topology=${topology-'[
     {"mem": "1G", "cores": 1, "nodes": 2, "packages": 2, "node-dist": {"4": 28, "5": 28}},
     {"nvmem": "8G", "node-dist": {"5": 28, "0": 17}},
```

Reply:
And while we are at it, IMO we better make it so that `create()` can also automatically drop into interactive upon failures.

Reply:
`on_verify`, `on_verify_fail`, `on_create`, `on_create_fail` hooks in place.